File: R/compute-advantage.R
compute_advantage <- function(states, rewards, V, gamma) {
  T <- length(rewards)
  advantages <- numeric(T)

  G <- 0.0
  for (t in rev(seq_len(T))) {
    G <- rewards[t] + gamma * G
    advantages[t] <- G - V[states[t]]
  }

  advantages
}

File: Julia/compute-advantage.jl
function compute_advantage(states, rewards, V, gamma)
    T = length(rewards)
    advantages = zeros(Float64, T)

    G = 0.0
    for t in T:-1:1
        G = rewards[t] + gamma * G
        advantages[t] = G - V[states[t]]
    end

    return advantages
end
